Web/Edubase.Web.UI/Areas/Establishments/Controllers/BulkCreateFreeSchoolsController.cs

```diff
@@ -128,7 +128,7 @@ private async Task<ActionResult> ResultInternalAsync(Guid id, BulkCreateFreeScho
             }
         }
 
-        private async Task<ActionResult> ResultInternalAjaxAsync(Guid id, BulkCreateFreeSchoolsViewModel viewModel, ApiResponse<BulkCreateFreeSchoolsResult> apiResponse)
+        private ActionResult ResultInternalAjaxAsync(Guid id, BulkCreateFreeSchoolsViewModel viewModel, ApiResponse<BulkCreateFreeSchoolsResult> apiResponse)
```

Comment:
I really struggled to figure out the original nested ternarys to figure out if the change is equivalent (I assume it is, just trying to rationalise it and be confident in my own mind).

If we're refactoring this piece of code perhaps we would consider removing the nested ternary operators in favour of simpler if statements? 

Personally I find the example below easier to reason about / follow at-a-glance, but appreciate it's somewhat a stylistic view.

_The below was produced using IDE refactorings ("convert to if" and "invert if") so I don't have any reason to doubt this being equivalent, but worth double checking for yourself before using as-is._

```csharp
        private ActionResult ResultInternalAjaxAsync(Guid id, BulkCreateFreeSchoolsViewModel viewModel, ApiResponse<BulkCreateFreeSchoolsResult> apiResponse)
        {
            var redirectUrl = string.Concat("/Establishments/bulk-create-free-schools/", id);
            if (!apiResponse.Success)
            {
                if (!apiResponse.HasErrors)
                {
                    throw new Exception("ApiResponse indicated failure, but no errors were supplied");
                }

                return Json(JsonConvert.SerializeObject(new {status = true, redirect = redirectUrl}));
            }

            viewModel.Result = apiResponse.GetResponse();
            if (viewModel.Result.IsProgressing())
            {
                return Json(JsonConvert.SerializeObject(new {status = false, redirect = redirectUrl}));
            }

            if (viewModel.Result.IsCompleted())
            {
                return Json(JsonConvert.SerializeObject(new {status = true, redirect = redirectUrl}));
            }

            throw new Exception($"The status of task {id} is unclear; the API did not provide a good enough response {Newtonsoft.Json.JsonConvert.SerializeObject(viewModel.Result)}");

        }
```

Reply:
There are two changes here.
The method signature - which removes the async and task part, as nothing async happens in the method

Then the if statement. As this is resolving IDE warnings and problems, I haven't widened the scope.
The bit changed gives the same result whether the result "HasCreatedEstablishments" or not - the JSON produced for both was exactly the same.

However much I would like to widen the scope of these changes, I want to keep a focus on only resolving IDE highlighted issues - otherwise I could be doing this for weeks.